support-frontend/assets/pages/[countryGroupId]/validation.ts

```diff
@@ -18,4 +18,5 @@ export function preventDefaultValidityMessage(
  * This uses a Unicode character class escape
  * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Unicode_character_class_escape
  */
-export const doesNotContainEmojiPattern = '^[^\\p{Emoji_Presentation}]+$';
+export const doesNotContainExtendedEmojiOrLeadingSpace =
+	'^[^\\p{Extended_Pictographic}\\s].[^\\p{Extended_Pictographic]*$';
```

Comment:
Is the `.` in here needed? I think because the `.` will match anything it could be a pictograph and that would pass the validation. E.g. "✈️" at the start wouldn't match, but if you put a valid character first it will "x✈️" (since the ✈️ matches the `.` character).

Reply:
Hmm , think` x🙂` validates correctly (or any position of emoji). Even with my typo? Isn't it the `Extended_Pictogram` that the issue here?

Reply:
> I think because regexps can be kind of hard to interpret maybe it'd be good to add some tests for it. Would be nice in the tests to have a list of examples of strings which are valid and strings which aren't.

100% agree that any conditional regex operations are prime candidates for testing, we've done this before when testing regexs used in the AB test set-up logic: https://github.com/guardian/support-frontend/blob/main/support-frontend/assets/helpers/abTests/__tests__/abtestTest.ts#L489

Reply:
@tjmw Thanks, regex testing overload I think have corrected no. Agree about tests, added a set of jest tests. Was just looking for the location of the tests, thanks @GHaberis